crates/fuel-gas-price-algorithm/gas-price-analysis/src/simulation.rs

```diff
@@ -137,13 +136,13 @@ impl Simulator {
             projected_cost_totals.push(updater.projected_total_da_cost);
 
             // Update DA blocks on the occasion there is one
-            if let Some(da_blocks) = &da_block {
-                let mut total_cost = updater.latest_known_total_da_cost_excess;
-                for block in da_blocks {
-                    total_cost += block.block_cost as u128;
-                    actual_costs.push(total_cost);
+            if let Some((range, cost)) = da_block {
+                for height in range.to_owned() {
+                    updater
+                        .update_da_record_data(height..(height + 1), *cost)
```

Comment:
```suggestion
                        .update_da_record_data(height..=height, *cost)
```

Reply:
Unfortunately, `RangeInclusive` isn't the same type as `Range`, I went back and forth on whether to do a generic for the few traits I needed or just do `Range` and I figured this was less noise.

Reply:
Willing to reconsider if generics seems better.